src/CLI.hs

```diff
@@ -201,11 +205,28 @@ runCLI args = handle handler $ do
       input <- readInput inputFile
       rules' <- getRules
       program <- parseProgram input inputFormat
-      rewritten <- rewrite' program rules' (RewriteContext program maxDepth maxCycles depthSensitive buildTerm must)
+      let saveFunc = createSaveStepFunc stepsDir outputFormat printMode omitListing omitComments input
+      rewritten <- rewrite' program rules' (RewriteContext program maxDepth maxCycles depthSensitive buildTerm must saveFunc)
       logDebug (printf "Printing rewritten 𝜑-program as %s" (show outputFormat))
       prog <- printProgram rewritten outputFormat printMode input
       output prog
       where
+        createSaveStepFunc :: Maybe FilePath -> IOFormat -> PrintMode -> Bool -> Bool -> String -> SaveStepFunc
```

Comment:
@yarkivaev I think most of the arguments of this function are unnecessary since they are available from `OptsRewrite`